evaluator/evaluator/routers/dataset.py

```diff
@@ -1,22 +1,260 @@
+                    "skill_type": "multiple-choice",
+                    "metric": "accuracy",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "choices": ["choice.text"],
+                        "answer_index": "1",
+                    },
+                },
+            },
+            "extractive_qa": {
+                "summary": "Extractive sample",
+                "description": "Dataset for Extractive question answering",
+                "value": {
+                    "dataset_name": "quoref",
+                    "skill_type": "extractive-qa",
+                    "metric": "squad",
+                    "mapping": {
+                        "id": "id",
+                        "question": "question",
+                        "context": "context",
+                        "answers": ["answers.text"],
+                    },
+                },
+            },
+        },
+    ),
+):
+    # check if the skill-type is not empty
+    # download the dataset and check if the dataset_name skill-type and metric exist on huggingface
+    logger.debug(f"dataset object: {dataset}")
+    try:
+        dataset_handler = DatasetHandler()
+        dataset_downloaded = dataset_handler.download_dataset(dataset.dataset_name)
+
+    except Exception as exception:
+        raise HTTPException(
+            404,
+            f"The Name of the dataset : {dataset.dataset_name} not found on huggingface!",
+        )
+
+    # Load the metric from HuggingFace
+    try:
+        load(dataset.metric)
+    except FileNotFoundError:
+        raise HTTPException(
+            404, f"Metric with name: {dataset.metric} not found on Huggingface"
+        )
+
+    # validate the skill_type if the skill_type
+    if dataset.skill_type != valid_skill_type[dataset.skill_type]:
+        raise HTTPException(
+            404,
+            f"skill_type: {dataset.skill_type} is not supported for our case!",
+        )
+
+        # Check if the dataset exist on the database, when yes, the item muss not be added
+    if mongo_client.client.evaluator.datasets.count_documents(
+        {"dataset_name": dataset.dataset_name}
+    ):
+        raise HTTPException(
+            200, f"dataset: {dataset.dataset_name} exist on the database "
+        )
+
+    else:
+
+        mongo_client.client.evaluator.datasets.insert_one(dataset.mongo())
+        # Download the dataset from
+        dataset_handler.download_dataset(dataset.dataset_name)
+        m = f"Dataset {dataset.dataset_name} has been added to the database!"
+        raise HTTPException(200, m)
+
+
+@router.get("/{dataset_name}", status_code=200)
+async def get_dataset(
+    dataset_name: str,
+):
+    # get the database
+    # get collection and find dataset_name on the collection
 
+    try:
+        # check if the item exist
+        if mongo_client.client.evaluator.datasets.count_documents(
```

Comment:
Musst du hier 2 Abfragen machen? Also erst (1) gucken ob es da ist und dann (2) abfragen? Die Abfrage (2) gibt dann ja einfach nichts zurück, wenn es das nicht gibt, dann kannst du dir (1) sparen.

Reply:
Der erste Abfrage prüft, ob der Eintrag in der Datenbank existiert. ok ich habe die 2. Abfrage gelöscht.

Reply:
ich habe eine gelöscht